client/catapult/src/catapult/cache/CacheChangesSerializer.h

```diff
@@ -25,6 +25,11 @@
 #include "catapult/io/Stream.h"
 #include <vector>
 
+#ifdef _MSC_VER
+#pragma warning(push)
+#pragma warning(disable : 4702) /* unreachable code */
+#endif
```

Comment:
@Jaguar0625 Do you know why this warning is not seen with Conan build? Will need to look into this. 🤔

Reply:
i don't ... i wonder what file(s) are including this when warning is raised 🤔

Reply:
It might be possible is something surfaced by the application of the parallel build flag (/MP) which is now enabled for MSVC. I will double check

Reply:
Actually also the Conan build (Windows + MSVC) triggers the error. Here the extract of the log

```
        C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.39.33519\include\stdfloat(12): warning STL4038: The contents of <stdfloat> are available only with C++23 or later. [D:\GitHub\symbol\client\catapult\build\tests\ca
       tapult\local\recovery\tests.catapult.local.recovery.vcxproj]
         C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.39.33519\include\stdfloat(12): warning STL4038: The contents of <stdfloat> are available only with C++23 or later. [D:\GitHub\symbol\client\catapult\build\tests\ca
       tapult\local\recovery\tests.catapult.local.recovery.vcxproj]
         C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.39.33519\include\stdfloat(12): warning STL4038: The contents of <stdfloat> are available only with C++23 or later. [D:\GitHub\symbol\client\catapult\build\tests\ca
       tapult\local\recovery\tests.catapult.local.recovery.vcxproj]
         D:\GitHub\symbol\client\catapult\src\catapult\cache\ChunkedDataLoader.h(54,1): warning C4702: unreachable code [D:\GitHub\symbol\client\catapult\build\tests\catapult\local\recovery\tests.catapult.local.recovery.vcxproj]
         D:\GitHub\symbol\client\catapult\src\catapult\cache\CacheChangesSerializer.h(39,1): warning C4702: unreachable code [D:\GitHub\symbol\client\catapult\build\tests\catapult\local\recovery\tests.catapult.local.recovery.vcxproj]
         D:\GitHub\symbol\client\catapult\src\catapult\cache\CacheChangesSerializer.h(60,1): warning C4702: unreachable code [D:\GitHub\symbol\client\catapult\build\tests\catapult\local\recovery\tests.catapult.local.recovery.vcxproj]
         D:\GitHub\symbol\client\catapult\src\catapult\cache\CacheChangesSerializer.h(61,1): warning C4702: unreachable code [D:\GitHub\symbol\client\catapult\build\tests\catapult\local\recovery\tests.catapult.local.recovery.vcxproj]
```

Which causes the build process to terminate with 
```
       "D:\GitHub\symbol\client\catapult\build\ALL_BUILD.vcxproj" (destinazione predefinita) (1) ->
       "D:\GitHub\symbol\client\catapult\build\tests\catapult\local\recovery\tests.catapult.local.recovery.vcxproj" (destinazione predefinita) (208) ->
       (destinazione: ClCompile) ->
         D:\GitHub\symbol\client\catapult\src\catapult\cache\ChunkedDataLoader.h(54,1): error C2220: the following warning is treated as an error [D:\GitHub\symbol\client\catapult\build\tests\catapult\local\recovery\tests.catapult.local.recovery
       .vcxproj]

    Warnings: 178
    Errors 1
```

Reply:
Maybe the issue has to do with MSVC.  I removed the disable of the unreachable code warning and built with  ``msbuild /p:Configuration=Release /p:Platform=x64 /m ALL_BUILD.vcxproj`` as documented and did not have any errors.  It might be interesting to see what the MSVC command is🤔

Reply:
Well `msbuild` actually **is** msvc: is a CLI shortcut for the same build command VS invokes from the UI.

TL;DR: The issue is related to MSVC builds in **Debug** mode.

It's been my fault not to specify I've been issuing `Debug` builds (not Release) also for conan.  @Wayonb test has been carried out for Release. And I can confirm this PR in Release mode does not trigger the unreachable code either. In Debug mode, instead, additional static analysis is performed over the executed code. This leads to the issue being triggered when tests are built in those cases when classes are initialized with literal constants: the analisys of the compiled test surfaces some paths are not reachable.

Example from ChunckedDataLoader.h:
```
		/// Loads the next data chunk of at most \a numRequestedEntries into \a destination.
		void next(uint64_t numRequestedEntries, typename TStorageTraits::DestinationType& destination) {
			numRequestedEntries = std::min(numRequestedEntries, m_numRemainingEntries);
			m_numRemainingEntries -= numRequestedEntries;
			while (numRequestedEntries--)
				TStorageTraits::LoadInto(TStorageTraits::Load(m_input), destination); // <-- Not reachable if m_numRemainingEntries is known at compile time to be zero.
		}
```

I've just changed the guard to constrain the warning suppression on debug builds only.

Reply:
@AndreaLanfranchi Thanks for the information.  I was also able to repo the build failure in Debug.  I opened an issue to look into this https://github.com/symbol/symbol/issues/1051